pkg/builder/kaniko/kaniko.go

```diff
@@ -0,0 +1,222 @@
+	watcher, err := k.K8sClientset.BatchV1().Jobs(k.K8sNamespace).Watch(ctx, metav1.ListOptions{
+		FieldSelector: fmt.Sprintf("metadata.name=%s", job.Name),
+	})
+	if err != nil {
+		return nil, ErrWatchingJob.Wrap(err)
+	}
+	defer watcher.Stop()
+
+	for {
+		select {
+		case event, ok := <-watcher.ResultChan():
+			if !ok {
+				return nil, ErrWatchingChannelCloseUnexpectedly
+			}
+
+			j, ok := event.Object.(*batchv1.Job)
+			if !ok {
+				continue
+			}
+
+			if j.Status.Succeeded > 0 || j.Status.Failed > 0 {
+				// Job completed (successfully or failed)
+				return j, nil
+			}
+		case <-ctx.Done():
+			return nil, ErrContextCancelled
+		}
+	}
+}
+
+func (k *Kaniko) firstPodFromJob(ctx context.Context, job *batchv1.Job) (*v1.Pod, error) {
+	podList, err := k.K8sClientset.CoreV1().Pods(k.K8sNamespace).List(ctx, metav1.ListOptions{
+		LabelSelector: fmt.Sprintf("job-name=%s", job.Name),
+	})
+	if err != nil {
+		return nil, ErrListingPods.Wrap(err)
+	}
+
+	if len(podList.Items) == 0 {
+		return nil, ErrNoPodsFound.Wrap(fmt.Errorf("job: %s", job.Name))
+	}
+
+	return &podList.Items[0], nil
+}
+
+func (k *Kaniko) containerLogs(ctx context.Context, pod *v1.Pod) (string, error) {
+	if len(pod.Spec.Containers) == 0 {
+		return "", ErrNoContainersFound.Wrap(fmt.Errorf("pod: %s", pod.Name))
+	}
+
+	containerName := pod.Spec.Containers[0].Name
+
+	logOptions := v1.PodLogOptions{
+		Container: containerName,
+	}
+
+	req := k.K8sClientset.CoreV1().Pods(k.K8sNamespace).GetLogs(pod.Name, &logOptions)
+	logs, err := req.DoRaw(ctx)
+	if err != nil {
+		return "", err
+	}
+
+	return string(logs), nil
+}
+
+func (k *Kaniko) cleanup(ctx context.Context, job *batchv1.Job) error {
+	err := k.K8sClientset.BatchV1().Jobs(k.K8sNamespace).
+		Delete(ctx, job.Name, metav1.DeleteOptions{
+			PropagationPolicy: &[]metav1.DeletionPropagation{metav1.DeletePropagationBackground}[0],
+		})
+	if err != nil {
+		return ErrDeletingJob.Wrap(err)
+	}
+
+	// Delete the associated Pods
+	err = k.K8sClientset.CoreV1().Pods(k.K8sNamespace).
+		DeleteCollection(ctx, metav1.DeleteOptions{}, metav1.ListOptions{
+			LabelSelector: fmt.Sprintf("job-name=%s", job.Name),
+		})
+	if err != nil {
+		return ErrDeletingPods.Wrap(err)
+	}
+
+	return nil
+}
+
+func DefaultCacheOptions(buildContext string) (*builder.CacheOptions, error) {
+	if buildContext == "" {
+		return nil, ErrBuildContextEmpty
+	}
+	hash := sha256.New()
+	_, err := hash.Write([]byte(buildContext))
+	if err != nil {
+		return nil, err
+	}
+	hashStr := hex.EncodeToString(hash.Sum(nil))
+
+	return &builder.CacheOptions{
+		Enabled: true,
+		Dir:     "",
+		Repo:    fmt.Sprintf("ttl.sh/%s:24h", hashStr),
+	}, nil
+}
+
+func prepareJob(b *builder.BuilderOptions) (*batchv1.Job, error) {
+	jobName, err := names.NewRandomK8(kanikoJobNamePrefix)
+	if err != nil {
+		return nil, ErrGeneratingUUID.Wrap(err)
+	}
+
+	oneInt32 := int32(1)
+	fiveInt32 := int32(5)
```

Comment:
any reason not to set these at top of the package with proper name, ie:

```go
var (
   DefaultParallelism = int32(5)
   BackoffLimit = int32(5)
)
```

Reply:
nice catch, will fix it in a minute